crates/nomination/src/tests.rs

```diff
@@ -0,0 +1,58 @@
+use crate::mock::*;
+use frame_support::assert_err;
+use vault_registry::{BtcPublicKey, VaultStatus, Wallet};
+
+fn dummy_public_key() -> BtcPublicKey {
+    BtcPublicKey([
+        2, 205, 114, 218, 156, 16, 235, 172, 106, 37, 18, 153, 202, 140, 176, 91, 207, 51, 187, 55, 18, 45, 222, 180,
+        119, 54, 243, 97, 173, 150, 161, 169, 230,
+    ])
+}
+
+fn register_vault(id: u64) {
+    <vault_registry::Pallet<Test>>::insert_vault(
+        &id,
+        vault_registry::Vault {
+            id,
+            to_be_replaced_tokens: 0,
+            to_be_issued_tokens: 0,
+            issued_tokens: 10,
+            to_be_redeemed_tokens: 0,
+            replace_collateral: 0,
+            backing_collateral: 100,
+            wallet: Wallet::new(dummy_public_key()),
+            banned_until: None,
+            status: VaultStatus::Active(true),
+        },
+    );
+}
+
+#[test]
+fn test_regular_vaults_are_not_opted_in_to_nomination() {
+    run_test(|| {
+        register_vault(BOB);
```

Comment:
IMO instead of adding a manually initialized vault to the list of vaults, this test would be better as an integration test where you call the `register_vault()` dispatchable function